jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -260,14 +265,22 @@ private Image pullBaseImage(
   }
 
   /**
-   * Looks through a manifest list for any amd64/linux manifest and downloads and returns the first
-   * manifest it finds.
+   * Looks through a manifest list for the user specified arch/os manifest and downloads and returns
+   * the first manifest it finds.
    */
-  private ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
+  @VisibleForTesting
+  ManifestAndDigest<?> obtainPlatformSpecificImageManifest(
       RegistryClient registryClient, V22ManifestListTemplate manifestListTemplate)
       throws RegistryException, IOException {
 
-    List<String> digests = manifestListTemplate.getDigestsForPlatform("amd64", "linux");
+    if (buildContext.getContainerConfiguration() != null) {
+      Set<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
```

Comment:
I was thinking of checking if the platforms set is empty here but i realized that our front end `ContainerBuildPlan ` does the job for us.Should I add a check or should I trust our api?

Reply:
Good question. Because `ContainerBuildPlan` never returns an empty list and we have added a test to ensure it's non-empty, in this case, I think I'm fine with assuming it's non-empty _given that `getContainerConfiguration()` is not null_.

We can certainly add more code to guard against an empty set and raise an error which may also be a good practice in general (depends on situations). But the integration between `ContainerBuildPlan` and the jib-core is pretty tight and and internal, I think assuming non-empty is OK _given that `getContainerConfiguration()` is not null_.`

That said, I noticed `getContainerConfiguration()` is `@Nullable`. I think we should figure out when this can be null and why.

Lastly, using streams works, but for just getting the first element, I'd go with
```java
    platforms.iterator().next();
```

Reply:
> That said, I noticed `getContainerConfiguration()` is `@Nullable`. I think we should figure out when this can be null and why.

See #2618 and #2619.

Reply:
Once #2619 is merged ,then I will need to adjust my code a little bit